### ISSUE TITLE
Getting error while obtaining a token

### ISSUE THREAD

**Issue by lalitgupta-blr** [Azure/azure-cli#16040]
## Describe the bug

**Command Name**
`az account get-access-token`

**Errors:**
```
'Response' object has no attribute 'status'
Traceback (most recent call last):
azure/cli/core/adal_authentication.py, ln 103, in set_token
    super(MSIAuthenticationWrapper, self).set_token()
python3.6/site-packages/msrestazure/azure_active_directory.py, ln 598, in set_token
    self.scheme, _, self.token = get_msi_token(self.resource, self.port, self.msi_conf)
python3.6/site-packages/msrestazure/azure_active_directory.py, ln 486, in get_msi_token
    result.raise_for_status()
python3.6/site-packages/requests/models.py, ln 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:50342/oauth2/token

...

azure/cli/core/adal_authentication.py, ln 114, in set_token
    .format(err.response.status, err.response.reason))
AttributeError: 'Response' object has no attribute 'status'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az account get-access-token --resource={} --query {} --output {}`

## Environment Summary
```
Linux-4.15.0-1098-azure-x86_64-with-debian-stretch-sid (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.14.0 *

Extensions:
ai-examples 0.2.4

```

**Comment by yonzhan** [comment]
account

**Comment by lalitgupta-blr** [comment]
I have opened case 2011250050000470 separately hence closing this one.

**Comment by jiasli** [comment]
The request to get a token from manage identity endpoint failed, but the error handling caused another failure. 

The bug in error handling should have been fixed in #15776. Please update to the latest version and try again, it will reveal the actual error.

**Comment by jiasli** [comment]
The original error maybe caused by a Cloud Shell limitation on `--resource` (#11386).
